src/acl/UrlPort.cc

```diff
@@ -14,6 +14,6 @@
 int
 ACLUrlPortStrategy::match(ACLData<MatchType> * &data, ACLFilledChecklist *checklist)
 {
-    return data->match(checklist->request->url.port());
+    return data->match(checklist->request->url.port().value_or(0));
```

Comment:
I have adjusted cf.data.pre to explicitly document that port zero in "port" ACL matches unknown ports. This is only a documentation improvement. The official ACL code (implicitly) works the same way.

Reply:
Not done. I disagree with some aspects of the suggested out-of-scope TODO and believe that planning this enhancement requires more discussion/research than we should do in this PR. Let's hash it out outside this PR if/when one of us starts working on `port` ACL enhancement.